Core/include/Acts/Navigation/NavigationState.hpp

```diff
@@ -80,16 +82,20 @@ struct NavigationState {
 
   /// That are the candidate surfaces to process
   SurfaceCandidates surfaceCandidates = {};
-  SurfaceCandidates::const_iterator surfaceCandidate = surfaceCandidates.cend();
+  std::size_t surfaceCandidateIndex = 0;
```

Comment:
So you're switching to index based candidates here. I understand why that's needed for #2625, but outside of the scope of that rewrite, does it have any benefits over the iterators?

Reply:
what's nice about indices is that the whole state gets trivially copyable